transformer_engine/pytorch/distributed.py

```diff
@@ -245,26 +245,59 @@ def in_fp8_activation_recompute_phase() -> bool:
     return _FP8_ACTIVATION_RECOMPUTE_PHASE
 
 
-def _get_active_autocast_contexts():
-    """
-    Returns new CPU and GPU torch.amp.autocast(..) contexts that match the active autocast state
-    at the time of this function's execution.
-    """
-    autocast_cached = torch.is_autocast_cache_enabled()
+TORCH_MAJOR = int(torch.__version__.split(".")[0])
+TORCH_MINOR = int(torch.__version__.split(".")[1])
+if TORCH_MAJOR == 2 and TORCH_MINOR >= 4:
```

Comment:
Same as previous comment. Better to have single function with version check inside.